src/tap/data/ResultSetTableIterator.java

```diff
@@ -731,7 +731,7 @@ else if (colValue instanceof Boolean)
 			else if (colType != null && colValue != null && colType.type == DBDatatype.CHAR && (colType.length == 1 || colType.length <= 0) && colValue instanceof String)
 				colValue = ((String)colValue).charAt(0);
 			// if the column value is a geometrical object, it must be serialized in STC-S:
-			else if (translator != null && colType != null && colType.isGeometry()){
+			else if ((translator != null && colType != null && colType.isGeometry()) || colValue instanceof byte[]) {
```

Comment:
For example, the one you proposed to Alberto in the other issue:

```
SELECT c, distance(c, point('', 187.48, 2.05))
FROM (SELECT TOP 10 centroid(s_region) AS c FROM ivoa.ObsCore) AS trick
```

in this case colType is "UNKNOWN".